### DISCUSSION TITLE
example of tensorboard logging + pytorch profiler

### DISCUSSION THREAD

**Discussion by bmahlbrand** [Lightning-AI/pytorch-lightning#7660] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Could anyone advise on how to use the Pytorch-Profiler plugin for tensorboard w/lightning's wrapper for tensorboard to visualize the results?

**Comment by bmahlbrand** [comment]
I think this is bugged, unless it's been fixed in the latest version

**Comment by tchaton** [comment]
Dear @bmahlbrand,

This is working. I tested it out last week :)

Here are the steps:

1. Use add `profiler=pytorch` to your Trainer as following
```
Trainer(profiler=`pytorch`)
```

or 

```
from pytorch_lightning.profilers import PyTorchProfiler
Trainer(profiler= PyTorchProfiler(...))
```

2. Install `pip install torch-tb-profiler`.

3. Run your script. Here is an example: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/profiler_example.py. Someone modified with the LightningCLI and I believe this doesn't work anymore ahah Just need to revert back to pure argparse.

4. This will generate traces in your `lightning_logs`. You can use tensorboard --logdir .../lightning_logs/version_{} to visualize them.
This should also work with multiple gpus.

Note. Please, use master as I resolved a silly bug last week :)

Best,
T.C

**Reply by bmahlbrand** [reply]
Having an intermittent issue where tensorboard's dashboard just spends forever loading and doesn't display any data, and when it does work, it loads forever on the profiler - not sure if it's lightning related or not though.
